File: core/src/main/java/co/ericp/freewayforecast/RouteForecastSource.java
package co.ericp.freewayforecast;

import java.util.List;

import rx.Single;

/**
 * The main API to our business logic.
 *
 * This takes a start and end location and departure time as parameters
 * and returns a list of routes, along with their weather conditions along the
 * way.
 */
public interface RouteForecastSource {
    Single<List<RouteForecast>> getRouteForecast(
            Location origin, Location destination, long departure);
}

File: core/src/main/java/co/ericp/freewayforecast/RouteForecastSource.kt
package co.ericp.freewayforecast

import co.ericp.freewayforecast.routes.Route
import io.reactivex.Observable

/**
 * The main API to our business logic.
 *
 * This takes a route and returns a collection of weather points as they become
 * available.
 */
interface RouteForecastSource {
    fun getRouteForecasts(routes: List<Route>,
                          departure: Long): Observable<RouteForecast>
}

